### PR TITLE
Use poise-python cookbook

### PR DESCRIPTION
**Pull request by gsreynolds**
Hi! Noticed #43 and just wanted to contribute poise-python based duplicity install we already had in our duplicity wrapper cookbook. Hope it helps!

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -33,13 +33,13 @@
...
-default['duplicity_ng']['source']['checksum'] = 'dec756ebe6cd25a6a55d9f31954fd8e07f4502063e4e148bdde4b1b3b2ad4615'
-default['duplicity_ng']['source']['version'] = '0.7.03'
+# default['duplicity_ng']['source']['checksum'] = 'dec756ebe6cd25a6a55d9f31954fd8e07f4502063e4e148bdde4b1b3b2ad4615'
+default['duplicity_ng']['source']['version'] = '0.7.10'
 default['duplicity_ng']['source']['url'] = "https://launchpad.net/duplicity/0.7-series/#{node['duplicity_ng']['source']['version']}/+download/duplicity-#{node['duplicity_ng']['source']['version']}.tar.gz" # rubocop:disable Metrics/LineLength
 
 default['duplicity_ng']['source']['gnupg']['version'] = '0.3.2'
 default['duplicity_ng']['source']['gnupg']['url'] = "http://downloads.sourceforge.net/project/py-gnupg/GnuPGInterface/#{node['duplicity_ng']['source']['gnupg']['version']}/GnuPGInterface-#{node['duplicity_ng']['source']['gnupg']['version']}.tar.gz" # rubocop:disable Metrics/LineLength
-default['duplicity_ng']['source']['gnupg']['checksum'] = '0ea672251e2e3f71b62fef0c01539519d500f6b338f803af6b57e67a73cca8e6'
+# default['duplicity_ng']['source']['gnupg']['checksum'] = '0ea672251e2e3f71b62fef0c01539519d500f6b338f803af6b57e67a73cca8e6'
```
Comment (chr4) [inline attributes/default.rb]: Please remove the commented out checksums

File: recipes/install_azure.rb [hunk 2]
```diff
@@ -20,12 +20,14 @@
 #
 
 # Install Azure-specific packages
-node['duplicity_ng']['source']['azure']['packages'].each { |n| package n }
+package node['duplicity_ng']['source']['azure']['packages'] do
```
Comment (chr4) [inline recipes/install_azure.rb]: Does Chef now support `package [array]`?

File: recipes/source.rb [hunk 3]
```diff
@@ -22,48 +22,48 @@
...
+
+  pip_requirements "#{Chef::Config[:file_cache_path]}/duplicity_requirements.txt" do
+    python python_bin
   end
-  python_pip 'paramiko'
-  python_bin = node['python']['binary']
 else
-  node['duplicity_ng']['source']['python']['packages'].each do |name|
-    package name
+  package node['duplicity_ng']['source']['python']['packages'] do
```
Comment (chr4) [inline recipes/source.rb]: See `package [array]` question above

**Comment by rmoriz** [comment]
Does the combination of package + pip addons work on centos?

**Comment by gsreynolds** [comment]
Thanks, happy to help out. I'll remove the comments and verify pip add-ons
tomorrow.

Chef has supported multipackage installs since 12.1 which was released 3
years ago, I thought it might be safe enough to swap over to the new
syntax. See https://docs.chef.io/resource_package.html#multiple-packages
and
https://blog.chef.io/2015/03/03/phil-dibowitz-chef-client-12-1-0-multipackage-installs/.
If you prefer, I think either the compat_resource or multipackage cookbooks
provide compat back to to 12.0.

**Comment by gsreynolds** [comment]
Hi, updated the PR now to address the requested changes. Verified as working under package + pip addons, source + pip addons and pip + pip addons for centos and debian.

Also added an install_par2 recipe for the Par2 Wrapper Backend which we are using `par2+scheme://[user[:password]@]host[:port]/[/]path`

The only issues I can see, not related to this PR, are that the `python-gnupginterface` package doesn't exist in debian > 7. (https://packages.debian.org/search?keywords=python-gnupginterface) and `python-swiftclient`can't be found under centos 7.

```
* No candidate version available for python-swiftclient
           ================================================================================
           Error executing action `install` on resource 'yum_package[python-swiftclient]'
           ================================================================================
```

**Comment by gsreynolds** [comment]
> Does the combination of package + pip addons work on centos?
Oops, just realised what you meant. Pushed a commit with another suite package-pip to cover this combination.

File: .kitchen.yml [hunk 4]
```diff
@@ -33,22 +31,54 @@ suites:
...
     - recipe[duplicity_ng::source]
+    - recipe[duplicity_ng::install_azure]
+    - recipe[duplicity_ng::install_boto]
+    - recipe[duplicity_ng::install_ftp]
+    - recipe[duplicity_ng::install_paramiko]
+    - recipe[duplicity_ng::install_par2]
+    - recipe[duplicity_ng::install_swift]
     attributes:
       duplicity_ng:
-        path: /usr/bin/duplicity
```
Comment (chr4) [inline .kitchen.yml]: Was there a reason to switch from the fully-qualified path to `duplicity`?

Comment (gsreynolds) [inline .kitchen.yml]: Yes, cross platform issues I mentioned in 72090c9 "... Fix duplicity paths under source suites, can either be /usr/bin/local/duplicity under debian family or /usr/bin/duplicity under rhel-ish." 

Either way, regardless of install method or platform, you would expect it to end up in the path?

**Comment by gsreynolds** [comment]
Thanks for approving @chr4 :+1:

File: .kitchen.yml (hunk 4 above)
Comment (chr4) [inline .kitchen.yml]: Makes sense.

**Comment by chr4** [comment]
LGTM, merging. Thanks for the contribution! Mind being added to the contributors list in the README?

**Comment by gsreynolds** [comment]
Don't mind at all, happy for my name and email on my profile to be listed.

**Comment by chr4** [comment]
Ok. Added you to the list of contributors, and released `2.0.0` on the Supermarket. I've chosen a major release for this.
Thanks again!

**Pull request merged**